odbc/src/api/functions.rs

```diff
@@ -2032,23 +2077,62 @@ pub unsafe extern "C" fn SQLSetConnectAttr(
 /// This is the WChar version of the SQLSetConnectAttr function
 ///
 /// # Safety
-/// Because this is a C-infereface, this is necessarily unsafe
+/// Because this is a C-interface, this is necessarily unsafe
 ///
 #[no_mangle]
 pub unsafe extern "C" fn SQLSetConnectAttrW(
-    _hdbc: HDbc,
-    _attr: ConnectionAttribute,
-    _value: Pointer,
-    _str_length: Integer,
+    connection_handle: HDbc,
+    attribute: ConnectionAttribute,
+    value_ptr: Pointer,
+    str_length: Integer,
 ) -> SqlReturn {
-    unimplemented!()
+    let mut err = None;
+    let conn_handle = MongoHandleRef::from(connection_handle);
+
+    // This scope is introduced to make the RWLock Guard expire before we write
+    // any error values via add_diag_info as RWLock::write is not reentrant on
+    // all operating systems, and the docs say it can panic.
+    let sql_return = {
+        let conn = must_be_valid!((*conn_handle).as_connection());
+        let mut conn_guard = conn.write().unwrap();
+
+        match attribute {
+            ConnectionAttribute::CurrentCatalog => {
+                let current_catalog =
+                    input_wtext_to_string(value_ptr as *const WChar, str_length as usize);
+                conn_guard.attributes.current_catalog = Some(current_catalog);
+                SqlReturn::SUCCESS
+            }
+            ConnectionAttribute::LoginTimeout => match FromPrimitive::from_u32(value_ptr as u32) {
+                Some(login_timeout) => {
+                    conn_guard.attributes.login_timeout = Some(login_timeout);
+                    SqlReturn::SUCCESS
+                }
+                None => {
+                    err = Some(ODBCError::InvalidAttrValue("SQL_ATTR_LOGIN_TIMEOUT"));
```

Comment:
```suggestion
                    err = Some(ODBCError::InvalidAttrValue(attribute));
```

Reply:
What about `ODBCError::UnsupportedConnectionAttribute(attribute)` below then? Does it mean that the error message is going to be "The feature {Attribute index} is not implemented"?
I don't see a test checking the error message itself, but this is what I'd expect.

Reply:
You are right, ODBCError::UnsupportedConnectionAttribute is the next line. Both are linked, but it explains the confusion!
Since  you said that when using ODBCError::InvalidAttrValue(attribute) it would use the int32 value from the enum and this was why you had to use the string value, I was asking if this was not also be a problem with ODBCError::UnsupportedConnectionAttribute(attribue).
This was originally a question because I did not know if there was something specific done for UnsupportedConnectionAttribute that I missed or not. You just answered this now. Thank you :) 
Can we make it so that it takes a string then?